tests/backup/backup_restore_basic_test.go

```diff
@@ -3189,3 +3190,145 @@ var _ = Describe("{DeleteNSDeleteClusterRestore}", func() {
+		})
+
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateApplicationClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			log.InfoD("Verifying cluster status for both source and destination clusters")
+			clusterStatus, err := Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", destinationClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", destinationClusterName))
+		})
+
+		Step("Taking alternate backups of application from source cluster to both S3 and NFS backup locations", func() {
+			log.InfoD("Taking alternate backups of application from source cluster to both S3 and NFS backup locations")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, bkpNamespaces)
+			for i := 0; i < numberOfAlternateBackups; i++ {
+				for locationUID, locationName := range backupLocationMap {
+					log.InfoD("Creating backup using the backup location of [%s]", locationName)
+					backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, RandomString(10))
+					backupNames = append(backupNames, backupName)
+					err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, locationName, locationUID, appContextsToBackup, labelSelectors, orgID, sourceClusterUid, "", "", "", "")
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+					log.InfoD("Verifying the type of backup")
+					//First backup for each backup location must be a full backup, rest should be incremental
+					if i == 0 {
+						err = IsFullBackup(backupName, orgID, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if backup [%s] is a full backup", backupName))
+					} else {
+						err = IsFullBackup(backupName, orgID, ctx)
+						if strings.Contains(err.Error(), "is an incremental backup") {
+							log.InfoD("The backup [%s] is an incremental backup", backupName)
+						} else {
+							log.FailOnError(err, "Issue with created backup.")
+						}
+					}
```

Comment:
IsFullBackup is not intended to verify whether the backup is incremental. How about implementing an IsIncrementalBackup function? It is not guaranteed that all subsequent backups will always be incremental. If you expect them to be incremental, how about using log.FailOnError or dash.VerifyFatal to handle cases where they are not?

Reply:
@kphalgun-px @ak-px need a little advise;
1. I didn't find the helper function 'IsIncrementalBackup' can you please guide me here?
2. Correct me if I'm wrong, as per the testcase we would not keep taking the backups as the number of alternate backups is something fixed. After the first backup, we will have 5 incremental backups. An addition check which can be done is instead of 'i==0', we can have 'i%6==0' ? 
3.  Krishna, I have handled the 'log.FailOnError' in the nested else. I'm trying to verify the full backup when its the first iteration, if its not the first then it must be an incremental, and if its a full backup instead of full backup or there is any issue the testcase will fail.

Reply:
You will have to implement IsIncrementalBackup

```
// IsIncrementalBackup checks if given backup is incremental backup or not
func IsIncrementalBackup(backupName string, orgID string, ctx context.Context) error {
	backupUid, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
	if err != nil {
		return err
	}
	backupInspectReq := &api.BackupInspectRequest{
		Name:  backupName,
		OrgId: orgID,
		Uid:   backupUid,
	}
	resp, err := Inst().Backup.InspectBackup(ctx, backupInspectReq)
	if err != nil {
		return err
	}
	for _, vol := range resp.GetBackup().GetVolumes() {
		backupId := vol.GetBackupId()
		log.Infof("BackupID of backup [%s]: [%s]", backupName, backupId)
		if strings.HasSuffix(backupId, "-incr") {
			return nil
		}
	}
	return fmt.Errorf("backup [%s] is an full backup", backupName)
}
```

I am not sure if the condition 'i % 6 == 0' will hold true when there is a lot of I/O activity in the namespace. Please confirm with someone else.

I missed the nested else part. My bad. Can you please make the error message more descriptive?

Reply:
I have made the error message more descriptive in the last commit.

Reply:
Thank you. Please include backup name as well.

Reply:
Sure Krishna, will do it.